android/src/main/java/com/reactnativesimcardsmanager/SimCardsManagerModule.java

```diff
@@ -172,10 +159,10 @@ public void setupEsim(ReadableMap config, Promise promise) {
       return;
     }
 
-    if (!checkCarrierPrivileges()) {
-      promise.reject("1", "No carrier privileges detected");
-      return;
-    }
+//    if (!checkCarrierPrivileges()) {
```

Comment:
@imranMnts could you elaborate why did you commented this please? For me the issue seems that the app has not in fact the carrier priviliges and it fails because of it later on.